src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemEventArgs.cs

```diff
@@ -17,11 +17,19 @@ public class FileSystemEventArgs : EventArgs
         /// </devdoc>
         public FileSystemEventArgs(WatcherChangeTypes changeType, string directory, string? name)
         {
+            ArgumentNullException.ThrowIfNull(directory);
+
             _changeType = changeType;
             _name = name;
+
+            if (string.IsNullOrEmpty(directory)) // empty directory means current working dir
```

Comment:
```suggestion
            if (directory == "") // empty directory means current working dir
```
`directory` was checked for null above. No need to do the check again